src/encoded/visualization.py

```diff
@@ -19,15 +26,196 @@
+        if not search_param_lists:
+            search_param_lists = {}
+
+    if 'File' in search_param_lists['type']:
+        # Add predefined sub-aggs to collect Exp and File counts from ExpSet items, in addition to getting own doc_count.
+
+        common_sub_agg = deepcopy(SUM_FILES_AGGREGATION_DEFINITION)
+
+        # Add on file_size_volume
+        for key_name in ['total_files']:
+            common_sub_agg[key_name + "_volume"] = {
+                "sum" : {
+                    "field" : common_sub_agg[key_name]["cardinality"]["field"].replace('.accession.raw', '.file_size')
+                }
+            }
+
+        if group_by_fields is not None:
+            group_by_agg_dict = {
+                group_by_field : {
+                    "terms" : {
+                        "field"     : "embedded." + group_by_field + ".raw",
+                        "missing"   : TERM_NAME_FOR_NO_VALUE,
+                        "size"      : 30
+                    },
+                    "aggs" : common_sub_agg
+                }
+                for group_by_field in group_by_fields if group_by_field is not None
+            }
+            histogram_sub_aggs = dict(common_sub_agg, **group_by_agg_dict)
+        else:
+            histogram_sub_aggs = common_sub_agg
+
+    else:
+        if group_by_fields is not None:
+            # Do simple date_histogram group_by sub agg, unless is set to 'None'
+            histogram_sub_aggs = {
+                group_by_field : {
+                    "terms" : {
+                        "field"     : "embedded." + group_by_field + ".raw",
+                        "missing"   : TERM_NAME_FOR_NO_VALUE,
+                        "size"      : 30
+                    }
+                }
+                for group_by_field in group_by_fields if group_by_field is not None
+            }
+        else:
+            histogram_sub_aggs = None
+
+    # Create an agg item for each interval in `date_histogram_intervals` x each date field in `date_histogram_fields`
+    # TODO: Figure out if we want to align these up instead of do each combination.
+    outer_date_histogram_agg = {}
+    for interval in date_histogram_intervals:
+        for dh_field in date_histogram_fields:
+            outer_date_histogram_agg[interval + '_interval_' + dh_field] = {
+                "date_histogram" : {
+                    "field": "embedded." + dh_field,
+                    "interval": interval_to_es_interval[interval],
+                    "format": "yyyy-MM-dd"
+                }
+            }
+            if histogram_sub_aggs:
+                outer_date_histogram_agg[interval + '_interval_' + dh_field]['aggs'] = histogram_sub_aggs
+
+    search_param_lists['limit'] = search_param_lists['from'] = [0]
+    subreq          = make_search_subreq(request, '{}?{}'.format('/search/', urlencode(search_param_lists, True)) )
+    search_result   = perform_search_request(None, subreq, custom_aggregations=outer_date_histogram_agg)
```

Comment:
You only need the aggregation result yes? Probably I should provide an endpoint to execute just an aggregation?